### PR TITLE
Apply P0663R0 PR for #259: is_swappable type traits should not be in …

### LINKED ISSUE #259: is_swappable type traits should not be in namespace std
They are defined in terms of unqualified calls to `swap`, but `swap` is unconstrained in C++14. Solution: move the `swappable` traits into the `std::experimental::ranges` namespace and define them in terms of the `ranges::swap` customization point object.

### PR DESCRIPTION
**Pull request by CaseyCarter**
…namespace `std`

Fixes #259.

### CODE REVIEW

File: utilities.tex [hunk 1]
```diff
@@ -787,6 +677,111 @@
...
+\endfirsthead
+\continuedcaption\\
+\topline
+\lhdr{Template} &   \chdr{Condition}    &   \rhdr{Precondition}    \\ \capsep
+\endhead
+
+\indexlibrary{\idxcode{is_swappable_with}}%
+\tcode{template <class T, class U>}\br
+  \tcode{struct is_swappable_with;} &
+  The expressions \tcode{ranges::swap(declval<T>(),}\brk\tcode{ declval<U>())}
```
Comment (ericniebler) [inline utilities.tex]: When I built this locally, the `ranges::swap(declval<T>(),` overran the middle column and partially obscured the text in the right column.

File: utilities.tex [hunk 2]
```diff
@@ -787,6 +677,111 @@
...
+  Only the validity of the immediate context of the \tcode{swap} expressions is
+  considered.
+  \enternote
+  The compilation of the expressions can result in side effects such as the
+  instantiation of class template specializations and function template specializations,
+  the generation of implicitly-defined functions, and so on. Such side effects are not
+  in the ``immediate context'' and can result in the program being ill-formed.
+  \exitnote &
+  \tcode{T} and \tcode{U} shall be complete types,
+  (possibly cv-qualified) \tcode{void}, or
```
Comment (ericniebler) [inline utilities.tex]: `\cv`

File: utilities.tex [hunk 3]
```diff
@@ -787,6 +677,111 @@
...
+  arrays of unknown bound.  \\ \rowsep
+
+\indexlibrary{\idxcode{is_swappable}}%
+\tcode{template <class T>}\br
+  \tcode{struct is_swappable;} &
+  For a referenceable type \tcode{T},
+  the same result as \tcode{is_\brk\-swappable_\brk\-with_\brk\-v<}\brk\tcode{T\&,\brk T\&>},
+  otherwise \tcode{false}. &
+  \tcode{T} shall be a complete type,
+  (possibly cv-qualified) \tcode{void}, or
```
Comment (ericniebler) [inline utilities.tex]: `\cv`

File: utilities.tex [hunk 4]
```diff
@@ -787,6 +677,111 @@
...
+  arrays of unknown bound. \\ \rowsep
+
+\indexlibrary{\idxcode{is_nothrow_swappable}}%
+\tcode{template <class T>}\br
+  \tcode{struct is_nothrow_swappable;} &
+  For a referenceable type \tcode{T},
+  the same result as \tcode{is_nothrow_swappable_}\brk\tcode{with_v<T\&, T\&>},
+  otherwise \tcode{false}. &
+  \tcode{T} shall be a complete type,
+  (possibly cv-qualified) \tcode{void}, or
```
Comment (ericniebler) [inline utilities.tex]: `\cv`

File: utilities.tex [hunk 5]
```diff
@@ -787,6 +677,111 @@
...
+
+\indexlibrary{\idxcode{is_nothrow_swappable_with}}%
+\tcode{template <class T, class U>}\br
+  \tcode{struct is_nothrow_swappable_with;} &
+  \tcode{is_swappable_with_v<T, }\brk\tcode{U>} is \tcode{true} and
+  each \tcode{swap} expression of the definition of
+  \tcode{is_swappable_with<T, U>} is known not to throw
+  any exceptions~(\cxxref{expr.unary.noexcept}). &
+  \tcode{T} and \tcode{U} shall be complete types,
+  (possibly cv-qualified) \tcode{void}, or
```
Comment (ericniebler) [inline utilities.tex]: `\cv`

File: utilities.tex [hunk 6]
```diff
@@ -787,6 +677,111 @@
 two \cv-unqualified non-reference types if at least one of them depends on a
 user-defined type. Such a specialization need not have a member named
 \tcode{type}.
+
```
Comment (ericniebler) [inline utilities.tex]: This whole block is showing up as blue, inserted text. It should be in black.

Comment (CaseyCarter) [inline utilities.tex]: Fix pending.

File: utilities.tex (hunk 1 above)
Comment (CaseyCarter) [inline utilities.tex]: Fix pending.

File: utilities.tex (hunk 2 above)
Comment (CaseyCarter) [inline utilities.tex]: Yip.

File: utilities.tex (hunk 3 above)
Comment (CaseyCarter) [inline utilities.tex]: Yip.

File: utilities.tex (hunk 5 above)
Comment (CaseyCarter) [inline utilities.tex]: Yip.

File: utilities.tex (hunk 4 above)
Comment (CaseyCarter) [inline utilities.tex]: You could have simply said "4 places" ;)

**Comment by CaseyCarter** [comment]
This should be fixed now.

**Pull request merged**